### PR TITLE
Tidy up dependencies

### LINKED ISSUE #428: Ensure packages don't duplicate dependencies
We should go through our package dependencies to make sure that transitive dependencies aren't duplicated in the current package. Also worth defining our own package versions in the dependencies, so we can be sure at release time that we have migrated versions correctly

> * Duplicated requirements for test and dev in https://github.com/UCLH-Foundry/PIXL/blob/main/cli/pyproject.toml and https://github.com/UCLH-Foundry/PIXL/blob/main/pixl_core/pyproject.toml and other pixl_* packages

_Originally posted by @mxochicale in https://github.com/UCLH-Foundry/PIXL/issues/422#issuecomment-2244609120_

### PR DESCRIPTION
**Pull request by p-j-smith**
## Description
Fixes #428 

- remove duplicated transitive dependencies and depend on `core` and `core` extras (`dev` and `test`) where appropriate

## Type of change
Please delete options accordingly to the description.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Suggested Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have passed on my local host device. (see further details at the [CONTRIBUTING](https://github.com/UCLH-Foundry/PIXL/blob/main/CONTRIBUTING.md#local-testing) document)
- [ ] Make sure your branch is up-to-date with main branch. See [CONTRIBUTING](https://github.com/UCLH-Foundry/PIXL/blob/main/CONTRIBUTING.md) for a general example to syncronise your branch with the `main` branch.
- [ ] I have requested PR review to `UCLH-Foundtry/arc-dev`
- [ ] I have adddressed and marked as resolved all the review comments in my PR.
- [ ] Finally, I have selected `squash and merge`

### CODE REVIEW

File: pixl_core/pyproject.toml [hunk 1]
```diff
@@ -7,36 +7,40 @@ readme = "README.md"
...
-    "token-bucket==0.3.0",
-    "python-decouple==3.8",
-    "python-slugify==8.0.4",
-    "pika==1.3.2",
-    "aio_pika==9.4.2",
-    "requests==2.32.2",
     "jsonpickle==3.2.2",
     "loguru==0.7.2",
-    "sqlalchemy==2.0.32",
-    "psycopg2-binary==2.9.9",
```
Comment (p-j-smith) [inline pixl_core/pyproject.toml:22]: I put the dependencies in alphabetical order to make it easier to see which ones are duplicated

Comment (stefpiatek) [reply pixl_core/pyproject.toml:22]: Yeah good call

**Review by stefpiatek — APPROVED** [review]
Looking good, thanks Paul!

File: cli/pyproject.toml [hunk 2]
```diff
@@ -14,14 +14,11 @@ dependencies = [
...
-    "pytest==8.3.*",
+    "core[test]==0.2.0rc",
     "pytest-mock==3.14.*",
-    "pytest-pixl"
 ]
 dev = [
-    "mypy",
-    "pre-commit",
-    "ruff",
+    "core[dev]==0.2.0rc",
```
Comment (mxochicale) [inline cli/pyproject.toml:21]: Do we need the same version as in ["0.2.0rc0"](https://github.com/UCLH-Foundry/PIXL/blob/main/pixl_core/pyproject.toml#L3C12-L3C20)?

**Review by mxochicale — APPROVED** [review]
Thanks Paul. It looks good to me, just added one comment!

File: cli/pyproject.toml (hunk 2 above)
Comment (p-j-smith) [reply cli/pyproject.toml]: oops, thanks for spotting that! I've fixed it now

**Comment by p-j-smith** [comment]
Thanks for the reviews!

**Pull request merged**